### PR TITLE
Prevent account name collisions

### LINKED ISSUE #15499: [Bug]: I am able to Create 2 Accounts with the same Name when anticipating pre-default Account name
### Describe the bug

**Problem**: if I create an Account with the name Account +` # of Accts+1` , when I follow up by creating another account I can see it will generate a duplicated Account name.
Usually I am not able to create 2 accounts with the same name.

https://user-images.githubusercontent.com/54408225/183404019-1faa7b46-71c8-417c-9252-b0fb9c882fd9.mp4

 

### Steps to reproduce

1. Create an Account with the name `Account ${# of Actts +1}`
2. Create another Account with the pre-deault name
Since the pre-default name is repeated with account created in 1, the result is that you will have 2 accounts with the same name.

### Error messages or log output

_No response_

### Version

10.19.0

### Build type

_No response_

### Browser

Chrome

### Operating system

Linux

### Hardware wallet

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by HowardBraham**
Fixes #15499

Goes a little beyond the bug report in 3 things:

- Works in any language.  So for instance if your MetaMask is set to Spanish, it does not let you create an account called "Cuenta 3"
- Fixed the localizations of `newAccountNumberName` in 4 locales: am, fa, no, and ta
- Stops you from renaming an existing account in the "Account Details" modal to a reserved name

@danjm @AlexJupiter

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -17,6 +17,8 @@ module.exports = {
     'node_modules/**/*',
     'jest-coverage/**/*',
   ],
+  // This is necessary to run eslint on Windows and not get a thousand CRLF errors
+  rules: { 'prettier/prettier': ['error', { endOfLine: 'auto' }] },
```
Comment (adonesky1) [inline .eslintrc.js:21]: can we separate this out to a different PR? Where we can have a separate discussion about this?

File: ui/components/ui/editable-label/editable-label.js [hunk 2]
```diff
@@ -19,91 +20,76 @@ class EditableLabel extends Component {
...
-  render() {
-    const { isEditing, value } = this.state;
-    const { className, accountsNames } = this.props;
-
     return (
-      <>
-        <div className={classnames('editable-label', className)}>
-          {isEditing ? this.renderEditing() : this.renderReadonly()}
+      <div className={classnames('editable-label', this.props.className)}>
+        <div key={1} className="editable-label__value">
```
Comment (adonesky1) [inline ui/components/ui/editable-label/editable-label.js:77]: what is the purpose of the key attributes here and below?

Comment (adonesky1) [reply ui/components/ui/editable-label/editable-label.js:77]: Oh I see, previously these components were in an array. Looks like they can be removed now?

File: ui/components/ui/editable-label/editable-label.js [hunk 3]
```diff
@@ -19,91 +20,76 @@ class EditableLabel extends Component {
...
-        key={2}
-        onClick={() => this.handleSubmit()}
-      >
-        <i className="fa fa-check editable-label__icon" />
-      </button>,
-    ];
+    return (
+      <div className={classnames('editable-label', this.props.className)}>
+        <input
+          key={1}
```
Comment (adonesky1) [inline ui/components/ui/editable-label/editable-label.js:43]: Looks like the `keys` here can be removed

File: ui/helpers/utils/accounts.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  const reservedRegEx = new RegExp(`^${localizedWordForAccount} \\d+$`, 'iu'); // Match strings starting with ${localizedWordForAccount} and then any numeral, case insensitive
+  const isReservedAccountName = reservedRegEx.test(newAccountName);
+
+  const isValidAccountName =
+    newAccountName === defaultAccountName || // What is written in the text field is the same as the placeholder
+    (!isDuplicateAccountName && !isReservedAccountName);
+
+  let errorMessage;
+  if (isValidAccountName) {
+    errorMessage = '\u200d'; // This is Unicode for an invisible character, so the spacing stays constant
```
Comment (adonesky1) [inline ui/helpers/utils/accounts.js:24]: can you explain what you mean about he spacing?

File: ui/pages/create-account/new-account.component.js [hunk 5]
```diff
@@ -69,16 +72,9 @@ export default class NewAccountCreateForm extends Component {
...
-                ' new-account-create-form__error',
-                ' new-account-create-form__error-amount',
-              )}
-            >
-              {this.context.t('accountNameDuplicate')}
-            </div>
-          ) : null}
+          <div className="new-account-create-form__error new-account-create-form__error-amount">
+            {errorMessage}
+          </div>
```
Comment (adonesky1) [inline ui/pages/create-account/new-account.component.js:77]: Do we want this div to exist even if there is no `errorMessage`?

Comment (adonesky1) [inline ui/pages/create-account/new-account.component.js:77]:
```suggestion
 { isValidAccountName ? 
          null : (
          <div className="new-account-create-form__error new-account-create-form__error-amount">
            {errorMessage}
          </div>
          )
     }
```

**Comment by adonesky1** [comment]
Looking good! Thanks for the clean up here! Left a few comments/questions. Also if you don't mind it can be helpful to add a video of the before/after behavior to give the reviewer an idea of the expected UX change.

File: ui/components/ui/editable-label/editable-label.js (hunk 2 above)
Comment (HowardBraham) [reply ui/components/ui/editable-label/editable-label.js:77]: Yes, you're right, the `key` can be removed.

File: ui/helpers/utils/accounts.js (hunk 4 above)
Comment (HowardBraham) [reply ui/helpers/utils/accounts.js:24]: If you return `null` or `''`, the dialog grows in height and the buttons move down when an error message appears (shown below).  This is also why I **do** want that div to exist even if there is no `errorMessage`.

image
image

**Comment by HowardBraham** [comment]
@adonesky1 Responded to your reviews and ready for another check

**Comment by montelaidev** [comment]
LGTM

**Review by montelaidev — DISMISSED** [review]

**Review by montelaidev — APPROVED** [review]

**Review by adonesky1 — APPROVED** [review]
LGTM! Great work Howard!
